src/integrationTest/groovy/CompilationAndGeneration.groovy

```diff
@@ -0,0 +1,110 @@
+import org.junit.*
+import static org.junit.Assert.assertTrue
+import static org.junit.Assert.fail
+import org.gradle.testfixtures.ProjectBuilder
+import org.gradle.api.Project
+import org.gradle.tooling.GradleConnector
+import org.gradle.tooling.ProjectConnection
+import org.gradle.tooling.BuildLauncher
+import java.util.Map
+import java.util.HashMap
+import cruise.umple.util.SampleFileWriter;
+
+public class CompilationAndGeneration {    
+    private static final DIRECTORIES_ROOT = "src/integrationTest/resources/TestProject/"
+    private static final Map<String, ArrayList<String>> DIRECTORIES_TO_CHECK
+    // Key = path to directory we want to check, value = the files we expect to find in the directory
+    static {
+        Map<String, ArrayList<String>> aMap = new HashMap<String, ArrayList<String>>()
```

Comment:
I like the immutable one more, immutability is always a good thing. 

Also, I don't think you quite understand the multimap, you use `<String, String>` and let it do the `Collection<? extends String>` because it can handle the data better than you by default. 

... I'm more just trying to harass you with Java best practices on this one, there was ultimately nothing wrong with your original one. Immutability is just king (it's why its even in Umple!).